planner/core/task.go

```diff
@@ -1256,6 +1256,11 @@ func (p *PhysicalLimit) sinkIntoIndexLookUp(t task) bool {
 	if !isTableScan {
 		return false
 	}
+	// We can't sink Limit into IndexLookUpReader when it scans a partition table in dynamic prune mode
+	// https://github.com/pingcap/tidb/issues/32516
+	if p.ctx.GetSessionVars().UseDynamicPartitionPrune() && ts.Table.Partition != nil {
```

Comment:
```suggestion
	if ts.Table.Partition != nil && !ts.isPartition {
```
the ts.isPartition should be set if the table is using static prune mode